### PR TITLE
Adds new label descriptions

### PR DESCRIPTION
**Pull request by AveshCSingh**
## What changes are proposed in this pull request?

Adds descriptions for some new labels to fill in existing gaps.

When this PR is LGTM'd, I will also create these labels.
## How is this patch tested?

N/A

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: ISSUE_TRIAGE.rst [hunk 1]
```diff
@@ -75,6 +75,7 @@ Components
 - ``area/models``: MLmodel format, model serialization/deserialization, flavors
 - ``area/projects``: MLproject format, project execution backends
 - ``area/scoring``: Local serving, model deployment tools, Spark UDFs
+- ``area/server``: MLflow server, UI server
```
Comment (harupy) [inline ISSUE_TRIAGE.rst:78]: @AveshCSingh What is the difference between `MLflow server` and `UI server`?

Comment (AveshCSingh) [reply ISSUE_TRIAGE.rst:78]: `MLflow server` (started with `mlflow server`) is the Python server that serves requests (eg `/runs/create`), while the `UI server` (started with `mlflow ui`) serves the tracking UI. Is this the right distinction / clear to contributors?

Comment (AveshCSingh) [reply ISSUE_TRIAGE.rst]: As discussed offline, the above is incorrect. `mlflow server` handles both API endpoints and serving the Javascript (via gunicorn), while the JS server is only used in development.

Thanks for your help understanding this! 

I've updated the PR to have a more clear area name, suggested by @sueann.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -26,6 +26,7 @@ Model Registry
 - [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
 - [ ] `area/projects`: MLproject format, project running backends
 - [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
+- [ ] `area/server-infra`: MLflow server, Javascript dev server
```
Comment (harupy) [inline .github/pull_request_template.md:29]:
```suggestion
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
```
nit

File: ISSUE_TRIAGE.rst [hunk 3]
```diff
@@ -75,6 +75,7 @@ Components
 - ``area/models``: MLmodel format, model serialization/deserialization, flavors
 - ``area/projects``: MLproject format, project execution backends
 - ``area/scoring``: Local serving, model deployment tools, Spark UDFs
+- ``area/server-infra``: MLflow server, Javascript dev server
```
Comment (harupy) [inline ISSUE_TRIAGE.rst:78]:
```suggestion
- ``area/server-infra``: MLflow server, JavaScript dev server
```
nit

**Review by harupy — APPROVED** [review]
LGTM!

File: .github/pull_request_template.md (hunk 2 above)
Comment (AveshCSingh) [reply .github/pull_request_template.md:29]: Thanks!

**Pull request merged**